### PR TITLE
[IOTDB-1253] [To Vector] Support null value in aligned time series

### PR DESCRIPTION
**Pull request by HTHou**
## Support the nullable aligned timeseries data by using bitmaps to recode the null value position of the tablet.

Some subtasks shows as below:

- [x] Use the bitmaps to indicate the null-value position in the tablet on the client side
- [x] Serialize and send the nullable tablet to the server side
- [x] Use the bitmaps in VectorTVList to show the null values in the memtable
- [x] Support nullable aligned timeseries value in TsFile (already done in Vector branch)
- [x] Support query memory data
- [x] Support query disk data
- [x] Add UT / IT
- [x] Fix serialize insertTablePlan with bitmap bug 

## Support inserting nullable aligned timeseries value with sql statement

```
IoTDB> insert into root.t1.wf01.wt01(time, (status, temperature)) values (4000, (true, 17.1))
Msg: The statement is executed successfully.
IoTDB> insert into root.t1.wf01.wt01(time, (status, temperature)) values (4001, (false, null))
Msg: The statement is executed successfully.
IoTDB> insert into root.t1.wf01.wt01(time, (status, temperature)) values (4002, (null, 3.9))
Msg: The statement is executed successfully.
IoTDB> select * from root
+-----------------------------+------------------------+-----------------------------+
|                         Time|root.t1.wf01.wt01.status|root.t1.wf01.wt01.temperature|
+-----------------------------+------------------------+-----------------------------+
|1970-01-01T08:00:04.000+08:00|                    true|                         17.1|
|1970-01-01T08:00:04.001+08:00|                   false|                         null|
|1970-01-01T08:00:04.002+08:00|                    null|                          3.9|
+-----------------------------+------------------------+-----------------------------+
Total line number = 3
It costs 0.207s
IoTDB> show timeseries
+-----------------------------+-----+-------------+--------+--------+-----------+----+----------+
|                   timeseries|alias|storage group|dataType|encoding|compression|tags|attributes|
+-----------------------------+-----+-------------+--------+--------+-----------+----+----------+
|     root.t1.wf01.wt01.status| null|      root.t1| BOOLEAN|     RLE|     SNAPPY|null|      null|
|root.t1.wf01.wt01.temperature| null|      root.t1|   FLOAT| GORILLA|     SNAPPY|null|      null|
+-----------------------------+-----+-------------+--------+--------+-----------+----+----------+
Total line number = 2
It costs 0.049s
```

### CODE REVIEW

File: server/src/main/java/org/apache/iotdb/db/utils/TypeInferenceUtils.java [hunk 1]
```diff
@@ -81,6 +81,8 @@ public static TSDataType getPredictedDataType(Object value, boolean inferType) {
           return floatingStringInferType;
         }
         // "NaN" is returned if the NaN Literal is given in Parser
+      } else if ("null".equals(strValue)) {
```
Comment (SilverNarcissus) [inline server/src/main/java/org/apache/iotdb/db/utils/TypeInferenceUtils.java:84]: What if user pass a "NULL" as value? Should we think "NULL" also a null?

File: session/src/main/java/org/apache/iotdb/session/Session.java [hunk 2]
```diff
@@ -1453,6 +1453,10 @@ private int calculateLength(List<TSDataType> types, List<Object> values)
   private void putValues(List<TSDataType> types, List<Object> values, ByteBuffer buffer)
       throws IoTDBConnectionException {
     for (int i = 0; i < values.size(); i++) {
+      if (values.get(i) == null) {
+        ReadWriteIOUtils.write((byte) -2, buffer);
```
Comment (SilverNarcissus) [inline session/src/main/java/org/apache/iotdb/session/Session.java:1457]: Maybe this "-2" can be a constant with a understandable name.

**Review by SilverNarcissus — CHANGES_REQUESTED** [review]
Nice work and abundant tests!  Only minor issues, please have a look~

File: session/src/main/java/org/apache/iotdb/session/Session.java (hunk 2 above)
Comment (HTHou) [reply session/src/main/java/org/apache/iotdb/session/Session.java:1457]: Sure

File: server/src/main/java/org/apache/iotdb/db/utils/TypeInferenceUtils.java (hunk 1 above)
Comment (HTHou) [reply server/src/main/java/org/apache/iotdb/db/utils/TypeInferenceUtils.java:84]: I'm going to let antlr transfer `NULL` to `null`. 
But this strValue may also come from session client by insertStringRecord, right? Yes, I think consider `NULL` here is necessary.

**Review by SilverNarcissus — APPROVED** [review]
Look good to me~  Only need to pass CI before we merge.

**Pull request merged**